src/Http/Offer/UpdateBookingAvailabilityRequestHandler.php

```diff
@@ -35,6 +36,7 @@ public function __construct(
 
     public function handle(ServerRequestInterface $request, string $offerId): ResponseInterface
     {
+        /** @var stdClass $data */
```

Comment:
Added this because PHPStan complained `Cannot access property $type on array|object.` in this file. Not sure why it complains when the return type is `array|object` as defined in the PSR7 standard for `getParsedBody()`, but does not when using `stdClass` (= an object).

Reply:
Okay apparently it also works if I just typehint it to `object`. Maybe because you cannot do `->` on an array. That would also explain why it doesn't complain in the other request handler where we use `RequestBodyParser`, because there the data is an array and we use `foreach ()` on it which works for both `array` and `object`. Not sure how to fix this easily without having to typehint/typecast in request handlers where an object is expected 🤔

Reply:
For now I just typecasted the return value to `(object)` in this request handler.

- If it's already an object (as it should be or the validation should throw) it will remain an object
- If it's an associative array for some reason, it will become an object. If the associative array had the right "properties", everything will keep working. If it didn't, the app will crash when an undefined property is accessed but it should do so anyway because the data was not correct then and the validation didn't catch it. So this should crash and get reported in Sentry and we should fix the validation then.
- If it's a sequential array for some reason, it will also become an object with properties for the sequential values. Of course the expected properties won't be set then, but again the app will crash as it should as soon as it tries to access one.

Overall I think this is just a part of the app where we would expect it to crash if we expect an object but get an array, because then the data was bad and the validation didn't catch it (= bug = internal server error). But of course PHPStan tries to warn us that we might be dealing with an array here, but there's not much we can do about it except add an `if (!is_object($data)) { throw ApiProblem::invalidBodyData(...); }` in every request handler where we expect an object but that's a lot for something that the validation layer should catch. Or we cast it and we "trick" PHPStan into thinking we're good.